pandas/core/indexes/base.py

```diff
@@ -139,7 +139,10 @@ def index_arithmetic_method(self, other):
 
 
 _o_dtype = np.dtype(object)
-_Identity = object
+
```

Comment:
does this actually change anything? why doesn't object work? why is this better?

Reply:
This is just to get better typing.

Reply:
I've updated the the PR to use [typing.NewType](https://docs.python.org/3/library/typing.html#newtype) instead of an actual class.